core/src/main/java/hudson/util/DirScanner.java

```diff
@@ -130,6 +139,51 @@ public void scan(File dir, FileVisitor visitor) throws IOException {
 
         private static final long serialVersionUID = 1L;
     }
+    
+    private static class DescendantFileSelector implements FileSelector{
+        private Set<String> alreadyDeselected;
```

Comment:
These fields can be `final` it looks like.

Reply:
Yeah they could, as there was no external usage, I was wondering. But you're right, I change it!